l1-contracts/src/core/libraries/crypto/SampleLib.sol

```diff
@@ -32,203 +41,67 @@ library SampleLib {
    *
    * @return indices - The indices of the committee
    */
-  function computeCommitteeStupid(uint256 _committeeSize, uint256 _indexCount, uint256 _seed)
+  function computeCommittee(uint256 _committeeSize, uint256 _indexCount, uint256 _seed)
     internal
-    pure
     returns (uint256[] memory)
   {
-    uint256[] memory indices = new uint256[](_committeeSize);
+    require(_committeeSize <= _indexCount, Errors.SampleLib__SampleLargerThanIndex(_committeeSize, _indexCount));
 
-    for (uint256 index = 0; index < _indexCount; index++) {
-      uint256 sampledIndex = computeShuffledIndex(index, _indexCount, _seed);
-      if (sampledIndex < _committeeSize) {
-        indices[sampledIndex] = index;
-      }
-    }
+    uint256[] memory sampledIndices = new uint256[](_committeeSize);
 
-    return indices;
-  }
-
-  /**
-   * @notice  Computing a committee slightly more cleverly.
-   *          Only computes for the committee size, and does not sample the full set.
-   *          This is more efficient than the stupid way, but still not optimal.
-   *          To be more clever, we can compute the `shuffeRounds` and `pivots` separately
-   *          such that they get shared accross multiple indices.
-   *
-   * @param _committeeSize - The size of the committee
-   * @param _indexCount - The total number of indices
-   * @param _seed - The seed to use for shuffling
-   *
-   * @return indices - The indices of the committee
-   */
-  function computeCommitteeClever(uint256 _committeeSize, uint256 _indexCount, uint256 _seed)
-    internal
-    pure
-    returns (uint256[] memory)
-  {
-    uint256[] memory indices = new uint256[](_committeeSize);
+    uint256 upperLimit = _indexCount - 1;
 
     for (uint256 index = 0; index < _committeeSize; index++) {
-      uint256 originalIndex = computeOriginalIndex(index, _indexCount, _seed);
-      indices[index] = originalIndex;
-    }
+      uint256 sampledIndex = computeShuffledIndex(index, upperLimit, _seed);
```

Comment:
Not sure I would call it a shuffle, but really, would probably just say sample so it also match you variable name 🤷